app/src/main/java/org/onionshare/android/ShareManager.kt

```diff
@@ -126,12 +126,15 @@ class ShareManager @Inject constructor(
                         _shareState.value =
                             ShareUiState.Starting(files, 100, state.progress)
                         onion = state.onion
+                    } else if (state is TorState.Stopping || state is TorState.Stopped) {
+                        // FIXME When Tor stops unplanned *after* starting,
+                        //  we also need to clear up our state, stopping webserver, etc.
+                        throw IOException("Tor has stopped unexpectedly")
```

Comment:
I was able to trigger this by repeatedly tapping the start/stop button - is that expected?

Reply:
That shouldn't happen anymore after merging the PR that disables the button and after the extra commits that refactor and improve the state handling. Now there's a dedicated STOPPING state with a disabled button which should also prevent this.